test/itkWaveletCoeffsPhaseAnalyzisImageFilterTest.cxx

```diff
@@ -126,53 +124,92 @@ itkWaveletCoeffsPhaseAnalyzisImageFilterTest( int argc, char *argv[] )
+      EXERCISE_BASIC_OBJECT_METHODS( waveletCoeffsPhaseAnalyzisImageFilter, WaveletCoeffsPhaseAnalyzisImageFilter,
+        ImageToImageFilter );
+
+      return runWaveletCoeffsPhaseAnalyzisImageFilterTest< 3, WaveletType >(  inputImage,
+        outputImage,
+        inputLevels,
+        inputBands,
+        applySoftThreshold,
+        thresholdNumOfSigmas );
+      }
     }
   else if ( waveletFunction == "Shannon" )
     {
     using WaveletType = itk::ShannonIsotropicWavelet< WaveletScalarType, ImageDimension>;
+      {
+      // Exercise basic object methods
+      // Done outside the helper function in the test because GCC is limited
+      // when calling overloaded base class functions.
+      using WaveletCoeffsPhaseAnalyzisImageFilterType = itk::WaveletCoeffsPhaseAnalyzisImageFilter<ImageType, WaveletType>;
+
+      auto waveletCoeffsPhaseAnalyzisImageFilter = WaveletCoeffsPhaseAnalyzisImageFilterType::New();
+      EXERCISE_BASIC_OBJECT_METHODS( waveletCoeffsPhaseAnalyzisImageFilter, WaveletCoeffsPhaseAnalyzisImageFilter,
+        ImageToImageFilter );
+
+      return runWaveletCoeffsPhaseAnalyzisImageFilterTest< 3, WaveletType >(  inputImage,
+        outputImage,
+        inputLevels,
+        inputBands,
+        applySoftThreshold,
+        thresholdNumOfSigmas );
+      }
     }
-  else
-    {
-    std::cerr << " failed!" << std::endl;
-    std::cerr << waveletFunction << " wavelet type not supported." << std::endl;
-    return EXIT_FAILURE;
-    }
-
-  // Exercise basic object methods
-  // Done outside the helper function in the test because GCC is limited
-  // when calling overloaded base class functions.
-  using WaveletCoeffsPhaseAnalyzisImageFilterType = itk::WaveletCoeffsPhaseAnalyzisImageFilter<ImageType, WaveletType>;
-
-  auto waveletCoeffsPhaseAnalyzisImageFilter = WaveletCoeffsPhaseAnalyzisImageFilterType::New();
-  EXERCISE_BASIC_OBJECT_METHODS( waveletCoeffsPhaseAnalyzisImageFilter, WaveletCoeffsPhaseAnalyzisImageFilter,
-  ImageToImageFilter );
-
-  if ( ImageDimension == 3 )
-    {
-    return runWaveletCoeffsPhaseAnalyzisImageFilterTest< 3, WaveletType >(  inputImage,
-                                                                            outputImage,
-                                                                            inputLevels,
-                                                                            inputBands,
-                                                                            applySoftThreshold,
-                                                                            thresholdNumOfSigmas );
-    }
-  else
-    {
-    std::cerr << "Test failed!" << std::endl;
-    std::cerr << "Error: only 2 or 3 dimensions allowed, " << dimension << " selected." << std::endl;
-    return EXIT_FAILURE;
-    }
+  std::cerr << " failed!" << std::endl;
+  std::cerr << waveletFunction << " wavelet type not supported." << std::endl;
+  return EXIT_FAILURE;
```

Comment:
Why the duplication of the code for each wavelet is needed?

Reply:
Not my code.  You would need to contact the original author.

Reply:
In your commit you are duplicating (4x) a chunk of code that @jhlegarreta or myself authored.

I am not asking about the original chunk of code, but about the reasons of duplication.

Maybe this is correct to avoid a shadowing of `WaveletType`, but just asking for your insight, if any, or maybe this is the result of an automatic refactor.